requirements/V2_REQUIREMENTS.md

```diff
@@ -0,0 +1,71 @@
+# WoT Profiles 2.0 Use Cases & Requirements
+
+## Introduction
+
+The [Web of Things](https://www.w3.org/WoT/) (WoT) seeks to counter the fragmentation of the [Internet of Things](https://en.wikipedia.org/wiki/Internet_of_things) (IoT) by using and extending existing, standardized web technologies.
+
+The [W3C WoT Thing Description 1.1 specification](https://w3c.github.io/wot-profile/#bib-wot-thing-description11) defines an information model and JSON-based representation format for describing the capabilities of connected devices and the interfaces with which to communicate with them. Thing Descriptions are designed to be protocol-agnostic and flexible enough to describe a wide range of existing ("brownfield") IoT devices.
+
+In order to provide this level of flexibility the Thing Description specification includes a number of extension points including protocol bindings, payload bindings, security mechanisms, link relations and semantic contexts. As long as all of the capabilities of a device can be described using a Thing Description and a [Consumer](https://w3c.github.io/wot-profile/#dfn-consumer) implements all of the extensions used, the Consumer should be able to interoperate with that device. However, the result of this extensible architecture is that any given Consumer can only interoperate with a subset of possible [Web Things](https://w3c.github.io/wot-profile/#dfn-thing).
+
+The WoT Profiles specification should complement the [Thing Description](https://w3c.github.io/wot-profile/#bib-wot-thing-description11) specification, by enabling ad-hoc interoperability through the use of "profiles". A profile prescribes a finite set of extensions and defaults that a Thing can be constrained to in order to guarantee out-of-the-box interoperability with any Consumer which implements that profile. Out-of-the-box interoperability means that a [Consumer](https://w3c.github.io/wot-profile/#dfn-consumer) which implements a given profile is guaranteed to be able to use every capability of a [Thing](https://w3c.github.io/wot-profile/#dfn-thing), which implements that same profile, without Thing-specific customization.
```

Comment:
@danielpeintner But this is exactly what it says. In particular, see points 8-13, one of which says that profiles "SHOULD NOT directly extend the WoT Thing Description information model".

When I say that Profiles are meant to complement the Thing Description specification, I just mean that they should not conflict with it. All of the requirements listed for profiles are about _constraining_ extension points, not about extending anything. However, describing a profile as being a "subset of the TD" is also not an accurate description, because it constrains much more than than just the Thing Description.

I feel you may be reacting to preconceived notions of what you think profiles are, or what has been discussed in the past, rather than the new approach described here and in the [strawman proposal](https://lists.w3.org/Archives/Public/public-wot-wg/2024Feb/0000.html).

Is there a way that this could be made clearer?

Reply:
@benfrancis Very good if we share the same understanding.
I tried to read it objectively. However, maybe I misunderstood some statements (I'm not a native speaker).

I have taken the liberty of suggesting some changes that would make it clearer to me.

```suggestion
The WoT Profiles specification should comply with the [Thing Description](https://w3c.github.io/wot-profile/#bib-wot-thing-description11) specification, by enabling ad-hoc interoperability through the use of "profiles". A profile prescribes defaults that a Thing can be constrained to in order to guarantee out-of-the-box interoperability with any Consumer which implements that profile. Out-of-the-box interoperability means that a [Consumer](https://w3c.github.io/wot-profile/#dfn-consumer) which implements a given profile is guaranteed to be able to use every capability of a [Thing](https://w3c.github.io/wot-profile/#dfn-thing), which implements that same profile, without Thing-specific customization.
```

On purpose I removed "finite set of extensions" since this means to me the opposite of what it should do. What do you see requiring being an extension?

Reply:
@danielpeintner OK, I think I understand what you are trying to clarify, but these edits don't quite make sense to me so let's try to refine them.

> The WoT Profiles specification should comply with the [Thing Description](https://w3c.github.io/wot-profile/#bib-wot-thing-description11) specification, by enabling ad-hoc interoperability through the use of "profiles".

It doesn't seem right to say that the Profiles specification "complies" with the Thing Description specification. It would be correct to say that the Thing Descriptions of Web Things which implement a Profile should comply with the Thing Description specification, but I had thought that was implied (and is now already the case with Profiles 1.0). Perhaps it needs to be made more explicit.

Also, the Profile specification does not enable ad-hoc interoperability simply by complying with the Thing Description specification, so this sentence doesn't really make sense.

> A profile prescribes ~a finite set of extensions and~ defaults that a Thing can be constrained to in order to guarantee out-of-the-box interoperability with any Consumer which implements that profile.

A profile doesn't just prescribe defaults, it constrains a Web Thing to using a particular set of extensions.

But when I say that a profile prescribes a finite set of extensions, I don't mean that it _defines_ those extensions (as Profiles 1.0 arguably does), it just lists them (see the [HTTP Basic Profile 2.0](https://docs.google.com/document/d/1LjBWiqQZXi85gXP2NNckQni5os6dwxW6UDGrZDb3cns/edit#heading=h.71drcnjjomtn) strawman proposal for an example . The "extensions points" are defined in the previous paragraph as being:
- Protocol bindings
- Payload bindings
- Security mechanisms
- Discovery mechanisms
- Link relations
- Semantic contexts

These are the existing extension points I have identified in the Web of Things which need to be constrained in order for a Consumer to guarantee interoperability. Essentially all a Profile should do in this proposal is to list the allowed extensions for each of these points, not define any new extensions.

--

To try to address your feedback I have:

Reply:
Thank you for the changes. It helps a lot. 

I was wondering whether we can also make it even more clear that profiles are meant to _use_ extension points and are never allowed to create *new* extensions (unless they go the usual way with for example creating a new binding or so etc)

Reply:
I think this is kind of already covered by points 8-12, but I've added some text to the introduction to reinforce this high level point.

> Profiles are designed to constrain, not extend, the Web of Things. They should only be used to constrain the set of options for existing extension points, never to extend the Web of Things directly.